### PR TITLE
changes for TOC BGMS upload validation

### PR DESCRIPTION
**Pull request by amit-tarento**

### CODE REVIEW

File: actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java [hunk 1]
```diff
@@ -731,6 +733,69 @@ private void throwInvalidDialCodeError(Set<String> invalidDialCodes) {
     return result;
   }
 
+  private void validateBGMSProperty(
+      int i,
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java]: i -> recordNum

File: actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java [hunk 2]
```diff
@@ -731,6 +733,69 @@ private void throwInvalidDialCodeError(Set<String> invalidDialCodes) {
...
+      Map<String, String> bgms,
+      HashMap<String, Object> recordMap,
+      Map<String, String> metadata) {
+    if (i == 0) {
+      if (recordMap.containsKey(JsonKey.BOARD)) {
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java]: Create a helper method.

File: actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java [hunk 3]
```diff
@@ -731,6 +733,69 @@ private void throwInvalidDialCodeError(Set<String> invalidDialCodes) {
...
+        bgms.put(JsonKey.GRADE, (String) recordMap.get(JsonKey.GRADE));
+      }
+      if (recordMap.containsKey(JsonKey.MEDIUM)) {
+        bgms.put(JsonKey.MEDIUM, (String) recordMap.get(JsonKey.MEDIUM));
+      }
+      if (recordMap.containsKey(JsonKey.SUBJECT)) {
+        bgms.put(JsonKey.SUBJECT, (String) recordMap.get(JsonKey.SUBJECT));
+      }
+    } else {
+      if (recordMap.containsKey(JsonKey.BOARD)
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java]: Create a helper method.

File: actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java [hunk 4]
```diff
@@ -731,6 +738,70 @@ private void throwInvalidDialCodeError(Set<String> invalidDialCodes) {
...
+      if (null != bgms.get(key) && null != recordMap.get(key)) {
+        String bgmsKey = ((String) bgms.get(key)).toLowerCase();
+        String recordMapKey = ((String) recordMap.get(key)).toLowerCase();
+        List<String> bgmsKeyList =
+            (Arrays.stream(bgmsKey.split(",")).map(s -> s.trim()).collect(Collectors.toList()));
+        List<String> recordMapKeyList =
+            (Arrays.stream(recordMapKey.split(","))
+                .map(s -> s.trim())
+                .collect(Collectors.toList()));
+        String[] bgmsArr = bgmsKeyList.toArray(new String[bgmsKeyList.size()]);
```
Comment (rahul-tarento) [inline actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/TextbookTocActor.java]: We can find the mismatch in streams and can avoid sorting and deepEquals

**Pull request merged**